### PR TITLE
Add an eslint config

### PR DESCRIPTION
**Pull request by davydog187**
Javascript is a quirky, interesting language that needs to be tamed :) I propose adding an eslint config that runs as part of the tests, to ensure that we're writing high quality javascript that avoids common gotchas.

This is an initial proposal, but the configuration is negotiable. I recommend turning on rules such as `no-const`, as well as some other rules that push authors towards functional javascript.

Thanks!

### CODE REVIEW

**Comment by davydog187** [comment]
The added `.eslintrc.json` was generated using `eslint --init` that inspects the existing javascript code.

File: assets/js/phoenix.js [hunk 1]
```diff
@@ -851,7 +855,7 @@ export class LongPoll {
     })
   }
 
-  close(code, reason){
```
Comment (davydog187) [inline assets/js/phoenix.js]: If the intention here was documented passed values, I can add the relevant jsdoc

File: assets/js/phoenix.js [hunk 2]
```diff
@@ -279,7 +279,7 @@ class Push {
     this.sent         = false
   }
 
-  matchReceive({status, response, ref}){
```
Comment (davydog187) [inline assets/js/phoenix.js]: same here

**Comment by Gazler** [comment]
This has been discussed before:

https://github.com/phoenixframework/phoenix/pull/2262
https://github.com/phoenixframework/phoenix/pull/2363

**Comment by davydog187** [comment]
Difference is that this adheres to the existing styles and only ensures consistency moving forward

**Comment by chrismccord** [comment]
Hey @davydog187 ,

Sorry this took so long for me to look at. For now I would prefer to wait on adding more JS tooling. I need less JS tooling in my life, not more :). That said, I absolutely see the desire and need for liners in the JS world and would use one on a large JS applications if I was in the setting. Thanks!

**Comment by davydog187** [comment]
Thanks chris!

**Pull request closed without merging**